src/components/FlowRunStateTypeCount.vue

```diff
@@ -0,0 +1,66 @@
+<template>
+  <div class="flow-run-state-type-count">
+    <span class="flow-run-state-type-count__badge">
+      <template v-for="state in states" :key="state">
+        <span class="flow-run-state-type-count__state" :class="getStateTypeClass(state)" />
+      </template>
+    </span>
+    <span class="flow-run-state-type-count__value">{{ count }}</span>
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import { computed } from 'vue'
+  import { useFlowRunsCount } from '@/compositions/useFlowRunsCount'
+  import { FlowRunsFilter } from '@/models/Filters'
+  import { StateType } from '@/models/StateType'
+  import { MaybeArray } from '@/types/utilities'
+  import { asArray } from '@/utilities/arrays'
+
+  const props = defineProps<{
+    stateType: MaybeArray<StateType>,
+    filter?: FlowRunsFilter,
+  }>()
+
+  const filter = computed(() => ({
+    ...props.filter,
+    flowRuns: {
+      ...props.filter?.flowRuns,
+      state: {
+        type: states.value,
+      },
+    },
+  }))
+
+  const states = computed(() => asArray(props.stateType))
+
+  const { count } = useFlowRunsCount(filter)
+
+  function getStateTypeClass(state: StateType): string {
+    return `bg-state-${state}-500`
+  }
```

Comment:
Don't we have a utility for this?

Reply:
I made that utility that gets the background and text color based on the `state--${stateType}` classes however I realized later (in that timeline light mode bug you fixed) that the background is different in light/dark mode. Which isn't really what we want here. This is something I'd like to revisit as a whole (state colors) but I know @brandonreid also wants to revisit them from a ux perspective. So I'm thinking we can tackle the devx and ux at the same time.

Reply:
Makes sense, thanks 👍🏻